sklearn/decomposition/tests/test_dict_learning.py

```diff
@@ -135,6 +145,25 @@ def test_dict_learning_online_shapes():
     assert_equal(np.dot(code, dictionary).shape, X.shape)
 
 
+def test_dict_learning_online_positivity():
+    rng = np.random.RandomState(0)
+    n_components = 8
+
+    dico = MiniBatchDictionaryLearning(
+        n_components, transform_algorithm='lasso_lars', random_state=0,
```

Comment:
If the test is not too long to run, it would be great to also check with other values of `transform_algorithm`.